language/move-vm/types/src/values/values_impl.rs

```diff
@@ -539,6 +578,20 @@ impl IndexedRef {
                 r1.borrow()[self.idx] == *r2.borrow()[other.idx].as_value_ref::<u8>()?
             }
 
+            (Locals(r1), VecU16(r2)) | (Struct(r1), VecU16(r2)) => {
```

Comment:
Huh... do these type mismatched cases ever happen? I don't think they should... but thats for another PR

Reply:
I actually dont know. I added them for completeness. 
Whats the ideal approach here?